torch/csrc/jit/ir/ir.h

```diff
@@ -1512,5 +1527,63 @@ struct OperatorSet {
   std::unordered_map<Symbol, std::vector<std::shared_ptr<Operator>>> ops;
 };
 
+template <typename T>
+struct OperatorMap {
+  // Type aliasing
+  using OpMapType = typename std::pair<std::shared_ptr<Operator>, T>;
+  using ValueType = std::vector<OpMapType>;
+  using MapType = std::unordered_map<Symbol, ValueType>;
+
+  OperatorMap() = default;
+  explicit OperatorMap(
+      std::initializer_list<std::pair<std::shared_ptr<Operator>, T>> init) {
+    insert(init);
+  }
+
+  void insert(const std::shared_ptr<Operator>& op, T val) {
+    map[Symbol::fromQualString(op->schema().name())].emplace_back(
+        std::make_pair(op, val));
+  }
+
+  void insert(
+      std::initializer_list<std::pair<std::shared_ptr<Operator>, T>> v) {
+    for (auto& el : v) {
+      insert(el.first, el.second);
+    }
+  }
+
+  void erase_all(const std::shared_ptr<Operator>& op) {
+    map.erase(Symbol::fromQualString(op->schema().name()));
+  }
+
+  bool contains(const Operator& op) const {
+    return map.count(Symbol::fromQualString(op.schema().name())) != 0;
+  }
+
+  c10::optional<ValueType> find(const Operator& op) {
+    if (!contains(op)) {
+      return c10::nullopt;
+    }
+    return map[Symbol::fromQualString(op.schema().name())];
+  }
+
+  // size and value_size used for testing
+  size_t size() const {
+    return map.size();
+  }
+
+  int64_t value_size(const std::shared_ptr<Operator>& op) const {
```

Comment:
i would leave off value_size as part of the public api because it's kind of confusing what it does and breaks the `Operator->T` abstraction